core/shared/platform/android/platform_internal.h

```diff
@@ -140,15 +140,15 @@ seekdir(DIR *__dir, long __location);
 
 #endif
 
-#if __ANDROID_API__ < 24
-
```

Comment:
This change is unrelated to the PR - just to fix an android build failure which has already been fixed on main. https://github.com/bytecodealliance/wasm-micro-runtime/pull/2594